pkg/tests/non-dependant/smcp_install_test.go

```diff
@@ -0,0 +1,107 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package non_dependant
+
+import (
+	_ "embed"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestSMCPInstall(t *testing.T) {
+	NewTest(t).Id("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			oc.DeleteNamespace(t, meshNamespace)
+			oc.CreateNamespace(t, meshNamespace)
+		})
+		rosa := strconv.FormatBool(env.IsRosa())
+		vars := map[string]string{
+			"Name":      smcpName,
+			"Namespace": meshNamespace,
+			"Rosa":      rosa,
+		}
+		versionTemplates := map[string]string{
+			"2.1": smcpV21_template,
+			"2.2": smcpV22_template,
+			"2.3": smcpV23_template,
+			// "2.4": smcpV24_template,
+		}
+		smcpVersion := env.GetDefaultSMCPVersion()
+		_, ok := versionTemplates[smcpVersion]
+		if !ok {
+			t.Fatal("Unsupported SMCP version: " + smcpVersion)
+		}
+		for version, smcpTemplate := range versionTemplates {
+			t.NewSubTest("smcp_test_install_" + version).Run(func(t TestHelper) {
+				t.LogStep("Delete Namespace, Create Namespace and Install SMCP v" + version)
+				oc.RecreateNamespace(t, meshNamespace)
+				installSMCPVersion(t, smcpTemplate, vars)
+				uninstallSMCPVersion(t, smcpTemplate, vars)
+			})
+		}
+		// Testing upgrade of SMCP to all supported version
+		versions := []string{"2.1", "2.2", "2.3", "2.4"}
+		for i := 0; i < len(versions)-1; i++ {
+			fromVersion := versions[i]
+			toVersion := versions[i+1]
+
+			t.NewSubTest(fmt.Sprintf("smcp_test_upgrade_%s_to_%s", fromVersion, toVersion)).Run(func(t TestHelper) {
+				oc.RecreateNamespace(t, meshNamespace)
+				installSMCPVersion(t, fromVersion, vars)
+				t.LogStep(fmt.Sprintf("Upgrade SMCP from v%s to v%s", fromVersion, toVersion))
+				installSMCPVersion(t, toVersion, vars)
+			})
+		}
+	})
+}
+func installSMCPVersion(t test.TestHelper, smcpTemplate string, vars interface{}) {
+	t.LogStep("Install SMCP")
+	oc.ApplyTemplate(t, meshNamespace, smcpTemplate, vars)
+	if env.IsRosa() {
+		oc.PatchWithMerge(
+			t, meshNamespace,
+			fmt.Sprintf("smcp/%s", smcpName),
+			`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
+	}
```

Comment:
This doesn't seem needed, as the template now handles this.

Reply:
Yes, my mistake for do not delete this